### ISSUE TITLE
The calculations do not  apply COLA increases accurately. 

### ISSUE THREAD

**Issue by bodzio4749** [Gregable/social-security-tools#180]
I love the site. I have created my own Excel spreadsheet as the  SSA.gov numbers are misleading, or hiding the monthly increases due.  The numbers that I came up with agree with the annual SSA projects to the penny, so (😜) I can claim some understanding of how they are / should be computed.

1. Your projected numbers exclude COLA increases that I dues since the age of 62, so assuming that on the year of full retirement the AIME number is increased only by the percentage of the retirement year, is wrong.  This underestimates the benefits due, and COLA increases should be compounded, not just added. In my case, it was 6.85% instead of 0.3% as your calculations show. Big difference. 
2.  It would be nice to have the calculation for individual months of planned retirement date. using a simple 8% per year increase assumes only that the retirement month can only happen on the month of the birthday.  SSA.gov seems to love this idea., but waiting extra 4-6 months instead of January of a year does not translate to a big difference in increases. 
3. Finally - I love your website. It aspires to be an excellent "second opinion" tool for all Americans. Even those not even thinking about their retirement years.

**Comment by Gregable** [comment]
I'm glad you found the site useful. Thanks for saying hello.

Re #1: The data is correct, but there are some assumptions that our calculations differ on. I'm not 100% sure what assumptions you are using, but a few things to keep in mind:

- The site doesn't include _future_ COLA adjustments. Those get [applied starting Jan 1](https://twitter.com/SsaTools/status/1448337245308551180) of the year they go into effect. The 2021 COLA goes into effect on Jan 1, 2022 which is in the future so the site don't apply it yet. 
- The site uses current indexing factors rather than projected  _future_ indexing factors. There is a detailed explanation of why [here](https://ssa.tools/guide/indexing-factors.html).

If these don't explain the differences, let me know.

Re #2: If you place your mouse over the chart immediately below the benefit adjustment table (which only shows years), you can get per-month adjustments. Otherwise the table would be rather long with 12 * 8 = 96 rows.

**Comment by bodzio4749** [comment]
#1:
I hate to disagree with you, but I must ... 😉. here [is the article](https://www.kiplinger.com/article/retirement/t051-c000-s004-delaying-social-security-boosts-the-value-of-colas.html) from  Kiplinger, which is a reliable source. Also, as I told you, my calculations agree with the once from SSA.GOV computed numbers to the penny.  Yes, for the majority of people the new COLA rate starts on January 1. There is only one exception - those who are born on Jan 1st, get it a month earlier.  from what I remember reading somewhere. 

- The AIME amount with compounded COLA adjustment is assumed for the FR age.  If you take retirement early, it is decreased by the appropriate percentage and can go down as much as 25% of AIME for folks like me with FRA at full 66.  A friend of mine has to wait extra 2 months to get full AIME (COLA adjusted).  
Notice, that the yearly amounts that you list calculate annual amounts starting with the full AIME amount, which is not correct. It should be reduced (in my age for example) by 25%, and then increased by COLA rate for age 62.  For age 63, the AIME should be reduced by 20%, etc... all the way to FRA where AIME would be full and updated by compounded  COLA since age 62. 

- As far as the future calculations are concerned, you are right: you cannot provide accurate predictions with only one exception so SSA.gov doesn't adjust them either:  SSA just announced a new COLA 5.9% adjustment for 2022 so you actually CAN adjust future numbers starting Jan 1, 2022, based on this already official number. I will let you know, but last December (or late November) all my SSA.gov numbers increased to reflect benefits as of Jan 1, 2021.  I will be watching them like this year too. I am still not postponing taking my SS benefits as I plan to convert as much money to ROTH IRA as possible before starting. 

#2:  For some reason, I don't seem to be able to find these monthly numbers. It's got to be my age. (⓿_⓿). A compromise, if I may suggest such, would be a link to a separate page with all the months listed. 

Thank you for your response. I am following you on Github and see if I can learn more python from your code. In a meantime, I am busy setting up my Raspberry Pi mini cluster at home.
